pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/PulsarAdminTool.java

```diff
@@ -43,121 +44,88 @@ public class PulsarAdminTool {
-        this.tlsTrustStoreType = properties.getProperty("tlsTrustStoreType", "JKS");
-        this.tlsTrustStorePath = properties.getProperty("tlsTrustStorePath");
-        this.tlsTrustStorePassword = properties.getProperty("tlsTrustStorePassword");
-
-        adminBuilder = PulsarAdmin.builder().allowTlsInsecureConnection(tlsAllowInsecureConnection)
+        return PulsarAdmin.builder().allowTlsInsecureConnection(tlsAllowInsecureConnection)
                 .enableTlsHostnameVerification(tlsEnableHostnameVerification)
                 .tlsTrustCertsFilePath(tlsTrustCertsFilePath)
                 .useKeyStoreTls(useKeyStoreTls)
                 .tlsTrustStoreType(tlsTrustStoreType)
                 .tlsTrustStorePath(tlsTrustStorePath)
                 .tlsTrustStorePassword(tlsTrustStorePassword);
+    }
 
-        jcommander = new JCommander();
-        jcommander.setProgramName("pulsar-admin");
-        jcommander.addObject(this);
-
-        commandMap = new HashMap<>();
-        commandMap.put("clusters", CmdClusters.class);
-        commandMap.put("ns-isolation-policy", CmdNamespaceIsolationPolicy.class);
-        commandMap.put("brokers", CmdBrokers.class);
-        commandMap.put("broker-stats", CmdBrokerStats.class);
-        commandMap.put("tenants", CmdTenants.class);
-        commandMap.put("resourcegroups", CmdResourceGroups.class);
-        commandMap.put("properties", CmdTenants.CmdProperties.class); // deprecated, doesn't show in usage()
-        commandMap.put("namespaces", CmdNamespaces.class);
-        commandMap.put("topics", CmdTopics.class);
-        commandMap.put("topicPolicies", CmdTopicPolicies.class);
-        commandMap.put("schemas", CmdSchemas.class);
-        commandMap.put("bookies", CmdBookies.class);
-
-        // Hidden deprecated "persistent" and "non-persistent" subcommands
-        commandMap.put("persistent", CmdPersistentTopics.class);
-        commandMap.put("non-persistent", CmdNonPersistentTopics.class);
-
-
-        commandMap.put("resource-quotas", CmdResourceQuotas.class);
-        // pulsar-proxy cli
-        commandMap.put("proxy-stats", CmdProxyStats.class);
-
-        commandMap.put("functions", CmdFunctions.class);
-        commandMap.put("functions-worker", CmdFunctionWorker.class);
-        commandMap.put("sources", CmdSources.class);
-        commandMap.put("sinks", CmdSinks.class);
-
-        // Automatically generate documents for pulsar-admin
-        commandMap.put("documents", CmdGenerateDocument.class);
-
-        // To remain backwards compatibility for "source" and "sink" commands
-        // TODO eventually remove this
-        commandMap.put("source", CmdSources.class);
-        commandMap.put("sink", CmdSinks.class);
-
-        commandMap.put("packages", CmdPackages.class);
-        commandMap.put("transactions", CmdTransactions.class);
+    protected void initRootParamsFromProperties(Properties properties) {
```

Comment:
not sure. It's a init method, not a reset one because the final state (after the method execution) is ready and not empty (after a reset method I expect a non initialized state)